core/src/Lender.sol

```diff
@@ -50,23 +50,32 @@ contract Lender is Ledger {
 
     constructor(address reserve) Ledger(reserve) {}
 
-    function initialize(IRateModel rateModel_, uint8 reserveFactor_) external {
+    function initialize() external {
         require(borrowIndex == 0);
         borrowIndex = uint72(ONE);
         lastAccrualTime = uint32(block.timestamp);
 
         initialDomainSeparator = _computeDomainSeparator();
         initialChainId = block.chainid;
+    }
 
+    /// @notice Sets the `rateModel` and `reserveFactor`. Only the `FACTORY` can call this.
+    function setRateModelAndReserveFactor(IRateModel rateModel_, uint8 reserveFactor_) external {
+        require(msg.sender == address(FACTORY));
         rateModel = rateModel_;
-        require(MIN_RESERVE_FACTOR <= reserveFactor_ && reserveFactor_ <= MAX_RESERVE_FACTOR);
```

Comment:
I know we check this in the factory, which is the only place this can be called, but it still feels weird not having a check in here for added safety (I know it costs a little extra gas).

Reply:
I agree it feels a little weird, but I want to keep all the `CONSTRAINT_XXX` constants in the Factory so that it's clear to people what exactly is allowed and what isn't. If we apply constraints in multiple places, that gets fuzzier.